### PR TITLE
Issue #4940: fix MultipleReturnPointsPerMethod idea violation

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4940

some of the cases looks weird or complicated, please comment what we should do with equals methods, and comment those hard cases that left unfixed.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java [hunk 1]
```diff
@@ -653,13 +654,14 @@ public FieldFrame getParent() {
          */
         private boolean isEmbeddedIn(String classOrEnumName) {
             FieldFrame currentFrame = this;
+            boolean isEmbeddedIn = false;
             while (currentFrame != null) {
                 if (Objects.equals(currentFrame.frameName, classOrEnumName)) {
-                    return true;
+                    isEmbeddedIn = true;
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java]: missing break after assignment.

**Comment by romani** [comment]
>  what we should do with equals methods

we should suppress vioalation on equals methods, as far as I remember IDEA inspection, have option for this, please activate.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java (hunk 1 above)
Comment (Nimfadora) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java]: fixed

File: src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java [hunk 2]
```diff
@@ -465,6 +465,7 @@ public static Configuration loadConfiguration(InputSource configSource,
      * @throws CheckstyleException if the string contains an opening
      *                           {@code ${} without a closing
      *                           {@code }}
+     * @noinspection MethodWithMultipleReturnPoints
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java]: this suppression is not allowed.
Please fix the code.
It has 3 returns, 1st will be ignored as it is guard, 2nd and 3rd have to be fixed. In this method should be two return points in result.

Comment (Nimfadora) [inline src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java]: fixed

**Pull request merged**